### ISSUE TITLE
Container::alias() is incorrectly named

### ISSUE THREAD

**Issue by mindplay-dk** [mindplay-dk/unbox#3]
After e.g. `$container->alias('B', 'A')`, overriding the registration of `B` has no effect on `A`.

In other words, the `alias()` method name is misleading, since this does not actually make `B` an _alias_ of `A` - the relationship isn't bi-directional, it's actually just `B` _referencing_ `A`.

This is perhaps part of the reason why statements such as `$container->alias('B', 'A')` can be difficult (for a person) to interpret - which is the created component registration, and which is the component being referenced by it?

Perhaps this would be more readable as e.g.:

``` php
$container->register('B', $container->ref('A'))
```

This is unambiguous, as clearly B is a reference to A.

In other words, add support for `BoxedValueInterface` as the second argument.

For backwards compatibility, maintain `alias()`, but flag as deprecated.

**Comment by mindplay-dk** [comment]
While this would be more readable as above, supporting this requires another overload of `register()`, where it accepts a boxed value - this complicates usage, as well as complicating things internally.

I'm going to preserve this method as-is.
